app/helpers/authentication_helper.rb

```diff
@@ -5,7 +5,7 @@ def authentication_provider(provider_name)
 
   def authentication_available_providers
     Authentication::Providers.available.map do |provider_name|
-      Authentication::Providers.const_get(provider_name.to_s.titleize)
+      Authentication::Providers.const_get(provider_name.to_s.camelize)
```

Comment:
Maybe it's time we stop having a helper reach into the guts of another class and instead add a method like `Authentication::Providers.available_providers` that either gets wrapped by this helper or used directly. This would also have the benefit of being able to wrap up the `name.to_s.camelize` logic in a single place in that class since I noticed you had to make two changes there too.

Reply:
Totally agree: the lack of ability to get the provider classes out of `Authentication::Providers` instead of just class names is really painful!